### PR TITLE
tests: Improve assertion error messages

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

**Review by yuwata — APPROVED** [review]

**Comment by DaanDeMeyer** [comment]
Added one more commit to introduce `ASSERT_OK_ERRNO()`, PTAL!

File: src/libsystemd/sd-bus/test-bus-watch-bind.c [hunk 1]
```diff
@@ -83,7 +83,7 @@ static void* thread_server(void *p) {
         assert_se(listen(fd, SOMAXCONN_DELUXE) >= 0);
         usleep_safe(100 * USEC_PER_MSEC);
 
-        assert_se(touch(path) >= 0);
```
Comment (mrc0mmand) [inline src/libsystemd/sd-bus/test-bus-watch-bind.c]: Hmm, `touch()` is our function, so it returns `-errno`?

Comment (daandemeyer) [reply src/libsystemd/sd-bus/test-bus-watch-bind.c]: Oh god I really thought it was a syscall, bad example

**Review by mrc0mmand — APPROVED** [review]

File: src/libsystemd/sd-bus/test-bus-watch-bind.c (hunk 1 above)
Comment (mrc0mmand) [reply src/libsystemd/sd-bus/test-bus-watch-bind.c]: To be honest I had to double check, since I thought that as well :)

**Review by yuwata — APPROVED** [review]

**Pull request merged**